text/0000-pack-direct-podman-call.md

```diff
@@ -0,0 +1,86 @@
+# Meta
+[meta]: #meta
+- Name: Pack direct podman call
+- Start Date: 2023-06-16
+- Author(s): dvaumoron
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+A flag `--container-engine` will be added to the `pack` CLI, the allowed flag value will be `docker` (the current functioning) and `podman` to call podman directly (without socket call)
+
+# Definitions
+[definitions]: #definitions
+
+N/A
+
+# Motivation
+[motivation]: #motivation
+
+This will allow to use pack as a standalone on the OS supported by podman (currently Linux, as Windows and macOS need podman machine)
+
+# What it is
+[what-it-is]: #what-it-is
+
+Adding `--container-engine podman` to the `pack` command which currently need docker call would allow them to work without docker installed.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The flag will change the initialization of the CommonAPIClient passed to call docker (interface defined in "github.com/docker/docker/client") by an adapter to call podman as a library. 
```

Comment:
Could we sketch out how the adapter would work?

Reply:
the goal is a to transmit/translate call to an initialized Runtime https://github.com/containers/podman/blob/v4.5.1/libpod/runtime.go#L63

Reply:
I can limit the adapter to the sub interface DockerClient https://github.com/buildpacks/pack/blob/v0.29.0/pkg/client/docker.go#L14, because the other method of the CommonAPIClient are not used within pack